src/group/random_perm.rs

```diff
@@ -0,0 +1,119 @@
+use crate::perm::Permutation;
+use rand::rngs::ThreadRng;
+use rand::Rng;
+use std::cmp::max;
+pub struct RandPerm {
+    size: usize,
+    rng: ThreadRng,
+    gen_elements: Vec<Permutation>,
+    a: Permutation,
```

Comment:
Following our discussion, it might be better to rename a to something like accum to improve readibility (I think we will make a big pull request afterwards in which we name stuff better)

Reply:
Yeah I'll rename that